src/main/java/io/jenkins/docker/connector/DockerComputerSSHConnector.java

```diff
@@ -253,8 +254,12 @@ protected ComputerLauncher createLauncher(DockerAPI api, String workdir, Inspect
         if ( retryWaitTimeOrNull!=null ) {
             connectionCheck.withEveryRetryWaitFor( retryWaitTimeOrNull, TimeUnit.SECONDS );
         }
+        final Integer launchTimeoutSeconds = getLaunchTimeoutSeconds();
+        if( launchTimeoutSeconds != null) {
+            connectionCheckSSH = connectionCheckSSH.withSSHTimeout(launchTimeoutSeconds, TimeUnit.SECONDS);
+        }
         final long timestampBeforeConnectionCheck = System.nanoTime();
-        if (!connectionCheck.execute() || !connectionCheck.useSSH().execute()) {
+        if (!connectionCheck.execute() || !connectionCheckSSH.execute()) {
             final long timestampAfterConnectionCheckEnded = System.nanoTime();
             final long nanosecondsElapsed = timestampAfterConnectionCheckEnded - timestampBeforeConnectionCheck;
             final long secondsElapsed = TimeUnit.NANOSECONDS.toSeconds(nanosecondsElapsed);
```

Comment:
As mentioned [earlier](https://github.com/jenkinsci/docker-plugin/pull/755#pullrequestreview-301209599)...

If we are intending to fix [JENKINS-59764](https://issues.jenkins-ci.org/browse/JENKINS-59764) then we should _also_ improve the exception message from [DockerComputerSSHConnector.createLauncher](https://github.com/jenkinsci/docker-plugin/pull/755/files#diff-08d87f09fabc2b54ea09327bb078ec28R266) so that we provide seconds and milliseconds rather than just seconds ... and we should also add text to the exception message saying something like `"Try increasing the number of retries (currently "+maxNumRetriesOrNull+") and/or the retry wait time (currently "+retryWaitTimeOrNull+") to allow for containers taking longer to start."` so that anyone encountering this error message will know that the answer is to increase the retries and/or wait time ... especially if either of them are null/zero.

i.e. Lets improve the exception message to ensure that nobody else is ever confused by this issue again.

Reply:
sure